scattering/van_hove.py

```diff
@@ -185,37 +137,27 @@ def compute_van_hove(
     return r, t, g_r_t_final
 
 
-def worker(return_dict, data):
-    key = (data[2], data[3])
-    r, g_r_t_partial = compute_partial_van_hove(*data)
-    return_dict[key] = g_r_t_partial
-    return_dict["r"] = r
-
-
-def compute_partial_van_hove(
-    trj,
-    chunk_length=10,
-    selection1=None,
-    selection2=None,
-    r_range=(0, 1.0),
-    bin_width=0.005,
-    n_bins=200,
-    self_correlation=True,
-    periodic=True,
-    opt=True,
-):
+
+def compute_partial_van_hove(trj, chunk_length=10, selection1=None, selection2=None, chunk_starts=None, 
+                             cpu_count=None, r_range=(0, 1.0), bin_width=0.005, n_bins=200, self_correlation=True, 
+                             periodic=True, n_concurrent_pairs=100000, opt=True, parallel=True):
+
     """Compute the partial van Hove function of a trajectory
 
     Parameters
     ----------
     trj : mdtraj.Trajectory
         trajectory on which to compute the Van Hove function
-    chunk_length : int
+    chunk_length : int, defualt=10
         length of time between restarting averaging
     selection1 : str
         selection to be considered, in the style of MDTraj atom selection
     selection2 : str
         selection to be considered, in the style of MDTraj atom selection
+    chunk_starts : int array-like, shape=(n_chunks,), optional, default=[chunk_length * i for i in range(trj.n_frames//chunk_length)]
+        The first frame of each chunk to be analyzed.
+    cpu_count : int, optional, default=min(multiprocessing.cpu_count(), total system memory in GB)
+        The number of cpu process to run at once if parallel is True
```

Comment:
The number of cpu *processes to simultaneously run if `parallel=True`.

Reply:
It get that. If a user passes a higher number for cpu_count than is available in the system, it just means more processes will be spawned by multiprocessing to compete for resources. That might cause a memory error or be slow if they pass a much bigger number, but other than that it shouldn't break anything. Depending on how the system handles threads, it may actually speed the code up on certain architectures.